connaisseur/alert.py

```diff
@@ -0,0 +1,177 @@
+            logging.error(err)
+        return response
+
+
+def load_config():
+    with open(
+        "{}/alertconfig.json".format(os.getenv("ALERT_CONFIG_DIR")), "r"
+    ) as configfile:
+        return json.loads(configfile.read())
+
+
+def get_images(admission_request):
+    relevant_spec = get_container_specs(
+        admission_request.get("request", {}).get("object", {})
+    )
+    return [container.get("image") for container in relevant_spec]
+
+
+def get_headers(receiver_config):
+    headers = {"Content-Type": "application/json"}
+    additional_headers = receiver_config.get("custom_headers")
+    if additional_headers is not None:
+        for header in additional_headers:
+            key, value = header.split(":", 1)
+            headers.update({key.strip(): value.strip()})
+    return headers
+
+
+def send_alerts(admission_request, *, admitted):
+    alert_config = load_config()
+    event_category = "admit_request" if admitted else "reject_request"
+    if alert_config.get(event_category) is not None:
+        for receiver in alert_config[event_category]["templates"]:
+            message = (
+                alert_config[event_category].get(
+                    "message", "CONNAISSEUR admitted a request."
+                )
+                if admitted
+                else alert_config[event_category].get(
+                    "message", "CONNAISSEUR rejected a request."
+                )
+            )
+            alert = Alert(message, receiver, admission_request)
+            alert.send_alert()
+
+
+def sanitize(template_dict, context):
+    if isinstance(template_dict, (dict, list)):
+        for key, value in template_dict.items():
+            if isinstance(value, dict):
+                sanitize(value, context)
+            elif isinstance(value, list):
+                for entry in value:
+                    sanitize(entry, context)
+            else:
+                template_dict[key] = Template(value).render(
+                    context, undefined=StrictUndefined
+                )
+    return template_dict
+
+
+def call_alerting_on_request(admission_request, *, admitted):
+    normalized_hook_image = Image(os.getenv("HELM_HOOK_IMAGE"))
+    hook_image = "{}/{}/{}:{}".format(
+        normalized_hook_image.registry,
+        normalized_hook_image.repository,
+        normalized_hook_image.name,
+        normalized_hook_image.tag,
+    )
+    images = []
+    for image in get_images(admission_request):
+        normalized_image = Image(image)
+        images.append(
+            "{}/{}/{}:{}".format(
+                normalized_image.registry,
+                normalized_image.repository,
+                normalized_image.name,
+                normalized_image.tag,
+            )
+        )
+    config = load_config()
+    templates = (
+        config.get("admit_request") if admitted else config.get("reject_request")
+    )
+    if images == [hook_image] or templates is None:
```

Comment:
Same as before. what if the config file has a wrong format? best add some basic input validation, should the config not be None to begin with.